### ISSUE TITLE
Docker entrypoint is run every time container is started

### ISSUE THREAD

**Issue by mstenta** [farmOS/farmOS#20]
I misunderstood the way that Docker's ENTRYPOINT scripts work - I thought they only ran when a container is initially created, but in fact they run every time the container is started. In hindsight, this makes more sense.

With this in mind, I will change the way that the entrypoint.sh works. Otherwise it will be rebuilding the farmOS codebase every time - which is both time consuming, and error prone.

**Comment by mstenta** [comment]
I changed the docker-entrypoint.sh so that it only rebuilds the farmOS codebase if index.php is not present. In other words, it only builds the code base once.

This solves the issues I've been running into with unwanted rebuilds. But it means that farmOS is not currently automatically updating.

I need to figure out a good way to do that.

For packaged release builds, I may be able to use Drush to compare the installed version of farmOS to the version in the Dockerfile - and rebuild if they don't match.

For dev builds, it's a little trickier... maybe I could have it run a `git pull` on the installation profile repository, and if there are new commits then run a rebuild. Or... maybe we shouldn't rebuild at all for development instances... and leave it up to the developer? We could include instructions for how to do that in the documentation...

**Comment by mstenta** [comment]
Alright I think I have a good solution to this.

First, I changed it so that rebuilds only occur if `profiles/farm/farm.info` is missing (rather than `index.php`). This provides a way to trigger a rebuild manually if you want to (without even destroying the container). I added a message to the output that describes how:

> To force a rebuild, delete profiles/farm/farm.info and restart the container.

This is how you rebuild if you have a development build.

For official release builds, I added some logic to detect the farmOS version in `profiles/farm/farm.info`.  If it doesn't match FARMOS_VERSION in the Dockerfile, then the codebase is replaced with the FARMOS_VERSION.

Here are the two relevant commits:

https://github.com/farmOS/farmOS/commit/047e3d324bac4f98b2216a81ca565d7ceccf9012

https://github.com/farmOS/farmOS/commit/fc9fbd886ce0e169e38c08c7ede24a3c11b6cf60

I'll update the documentation at http://farmos.org/development/docker/ to reflect this.
